src/java/htsjdk/samtools/sra/SRALazyRecord.java

```diff
@@ -0,0 +1,941 @@
+        super.setAttribute(tag, value);
+    }
+
+    @Override
+    protected void setAttribute(final short tag, final Object value, final boolean isUnsignedArray) {
+        AttributeTagInfo info = lazyAttributeTags.get(tag);
+        if (info != null && !initializedAttributes.contains(info.lazyAttribute)) {
+            initializedAttributes.add(info.lazyAttribute);
+        }
+        super.setAttribute(tag, value, isUnsignedArray);
+    }
+
+    @Override
+    public void clearAttributes() {
+        for (LazyAttribute lazyAttribute : LazyAttribute.values()) {
+            if (!initializedAttributes.contains(lazyAttribute)) {
+                initializedAttributes.add(lazyAttribute);
+            }
+        }
+        super.clearAttributes();
+    }
+
+    @Override
+    protected void setAttributes(final SAMBinaryTagAndValue attributes) {
+        for (LazyAttribute lazyAttribute : LazyAttribute.values()) {
+            if (!initializedAttributes.contains(lazyAttribute)) {
+                initializedAttributes.add(lazyAttribute);
+            }
+        }
+        super.setAttributes(attributes);
+    }
+
+    @Override
+    protected SAMBinaryTagAndValue getBinaryAttributes() {
+        for (Map.Entry<Short, AttributeTagInfo> info : lazyAttributeTags.entrySet()) {
+            if (!initializedAttributes.contains(info.getValue().lazyAttribute)) {
+                getAttribute(info.getKey());
+            }
+        }
+
+        return super.getBinaryAttributes();
+    }
+
+    public boolean isUnsignedArrayAttribute(final String tag) {
+        Short binaryTag = SAMTagUtil.getSingleton().makeBinaryTag(tag);
+        AttributeTagInfo info = lazyAttributeTags.get(binaryTag);
+        if (info != null && !initializedAttributes.contains(info.lazyAttribute)) {
+            getAttribute(binaryTag);
+        }
+
+        return super.isUnsignedArrayAttribute(tag);
+    }
+
+    // ===== misc ====
+
+    @Override
+    public boolean equals(final Object o) {
+        throw new RuntimeException("comparing of SRALazyRecords is not implemented yet");
+    }
+
+    @Override
+    public int hashCode() {
+        throw new RuntimeException("hashing of SRALazyRecords is not implemented yet");
+    }
+
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        throw new RuntimeException("cloning of SRALazyRecords is not implemented yet");
+    }
+
+    @Override
+    public String format() {
+        if (!initializedAttributes.contains(LazyAttribute.RG)) {
+            getAttribute("RG");
+        }
+        return super.format();
+    }
+
+    @Override
+    public List<SAMValidationError> isValid(final boolean firstOnly) {
+        if (!initializedFields.contains(LazyField.REFERENCE_NAME)) {
+            getReferenceName();
+        }
+
+        if (!initializedFields.contains(LazyField.MATE_REFERENCE_NAME)) {
+            getMateReferenceName();
+        }
+
+        getFlags();
+
+        return super.isValid(firstOnly);
```

Comment:
As currently written, `isValid()` will not fully validate the data in the record, since some fields (like the base qualities, etc.) may not have been lazily initialized yet.